Storage/rest/structured.h

```diff
@@ -73,14 +78,26 @@ struct Structured {
   // Returns `false` on error.
   static bool ExtractIfUnmodifiedSinceOrRespondWithError(Request& request,
                                                          Optional<std::chrono::microseconds>& destination) {
-    if (request.headers.Has("If-Unmodified-Since")) {
-      const auto& header_value = request.headers.Get("If-Unmodified-Since");
+    // If `X-Current-If-Unmodified-Since` header is set, traditional `If-Unmodified-Since` is ignored.
```

Comment:
Nit: if **the** `X-Current-If-Unmodified-Since` header is set, **the** traditional ...